Sources/Controllers/Map/Layers/OAGPXLayer.mm

```diff
@@ -154,26 +161,29 @@ - (void) refreshGpxTracks
         {
             if (it.key().isNull() || !it.value())
                 continue;
-            
+
             BOOL routePoints = NO;
-            
+
             OAGPX *gpx = [self getGpxItem:it.key()];
+            NSString *path = [self.app.gpxPath stringByAppendingPathComponent:gpx.gpxFilePath];
```

Comment:
Why do you need to create an OAGPXDocument each time if it's only needed fro the colorization?